### PR TITLE
Vijith/feat/editor font size

### LINKED ISSUE #8: Font Selector
**Is your feature request related to a problem? Please describe.**
I would like the user to be able to select the font size of the code

**Describe the solution you'd like**
There should be a dropdown that let's user to select font sizes xs, sm, md, lg, xl etc

**Describe alternatives you've considered**
Even a slider would be fine

### PR DESCRIPTION
**Pull request by vijith07**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +52,6 @@
   },
   "engines": {
     "node": ">=16.17.0",
-    "npm": ">=8.19.1"
+    "npm": ">=8.15.0"
```
Comment (notshriram) [inline package.json]: Ello Ello Ello. What's all this, then?

File: src/components/EditorSettings.tsx [hunk 2]
```diff
@@ -54,69 +78,76 @@ const EditorSettings = ({
...
-                            ))}
-                          </RadioGroup>
-                        </Disclosure.Panel>
-                      </Transition>
-                    </>
-                  )}
-                </Disclosure>
-              )}
-            </Menu.Item>
+                        {({}: any) => (
```
Comment (notshriram) [inline src/components/EditorSettings.tsx]: As mentioned previously, `any` should almost never be used. Here too it can be avoided.

**Review by notshriram — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (notshriram) [reply package.json]: @akashdoppalapudi do you want to downgrade to 8.15.0?

Comment (akashdoppalapudi) [reply package.json]: @notshriram Let's have latest version 8.19.1

File: src/components/EditorSettings.tsx (hunk 2 above)
Comment (akashdoppalapudi) [reply src/components/EditorSettings.tsx]: If possible... Let's use interfaces.
let's take advantage of typescript

File: package.json (hunk 1 above)
Comment (vijith07) [reply package.json]: > @notshriram Let's have latest version 8.19.1

why ?

File: src/components/EditorSettings.tsx [hunk 3]
```diff
@@ -32,11 +34,33 @@ const EditorSettings = ({
...
+    // "text-9xl"
   ];
+  const fontSizes = {
+    "text-xs": "12px",
+    "text-sm": "14px",
+    "text-base": "16px",
+    "text-lg": "18px",
+    "text-xl": "20px",
+    "text-2xl": "22px",
+    "text-3xl": "24px",
```
Comment (notshriram) [inline src/components/EditorSettings.tsx:51]: This does not necessarily correspond to the px values mentioned in the right side.

From [mdn](https://developer.mozilla.org/en-US/docs/Web/CSS/length)
> Represents the [font-size](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size) of the root element (typically [<html>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/html)). When used within the root element [font-size](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size), it represents its initial value (a common browser default is 16px, but user-defined preferences may modify this).

So depending on the browser settings, 1 rem could be != 16px

File: src/contexts/ThemeContext.tsx [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
 export const ThemeProvider = ({ children }: any) => {
   const isDarkPreferred = window.matchMedia(
     "(prefers-color-scheme: dark)"
   ).matches;
-  const [theme, setTheme] = useState(isDarkPreferred ? "dark" : "light");
+
+  const [theme, setTheme] = useLocalStorage(
+    "theme",
+    isDarkPreferred ? "dark" : "light"
+  );
```
Comment (notshriram) [inline src/contexts/ThemeContext.tsx:14]: 👍

File: package.json (hunk 1 above)
Comment (notshriram) [reply package.json]: > > @notshriram Let's have latest version 8.19.1
> 
> why ?

https://github.com/npm/cli/blob/latest/CHANGELOG.md#8191-2022-09-01

**Review by notshriram — APPROVED** [review]

**Pull request merged**